src/sql/workbench/common/actions.ts

```diff
@@ -239,16 +239,23 @@ export class BackupAction extends Task {
 	}
 
 	runTask(accessor: ServicesAccessor, profile: IConnectionProfile): void | Promise<void> {
```

Comment:
Unrelated to your PR...but is returning void | Promise<void> actually something people do? That seems really messy and makes handling the return type difficult - why wouldn't we just wrap a void return in a Promise.resolve instead?